web/src/features/charts/hooks/useBreakdownChartData.ts

```diff
@@ -37,20 +37,19 @@ export const getLayerFill = (
 export default function useBreakdownChartData() {
   const { data: zoneData, isLoading, isError } = useGetZone();
   const co2ColorScale = useCo2ColorScale();
+  const { zoneId } = useParams();
   const [mixMode] = useAtom(productionConsumptionAtom);
   const [displayByEmissions] = useAtom(displayByEmissionsAtom);
   const [aggregateToggle] = useAtom(spatialAggregateAtom);
   const isAggregateToggled = aggregateToggle === ToggleOptions.ON;
-  const [selectedDatetime] = useAtom(selectedDatetimeIndexAtom);
-  const currentData = zoneData?.zoneStates?.[selectedDatetime.datetimeString];
-  if (isLoading || isError || !currentData) {
+  if (isLoading || isError || !zoneData || !zoneId) {
     return { isLoading, isError };
   }
 
   const exchangesForSelectedAggregate = getExchangesToDisplay(
-    currentData.zoneKey,
+    zoneId,
     isAggregateToggled,
-    currentData.exchange
+    zoneData?.zoneStates
```

Comment:
Same goes here, just adding a comment so we don't forget this one if we change the other one.